src/apis/setColorPalette.js

```diff
@@ -1,14 +1,61 @@
-import actions from 'actions';
-import { getDataWithKey, mapToolNameToKey } from 'constants/map';
-import mapPaletteToAnnotationColorProperty from 'constants/mapPaletteToAnnotationColorProperty';
-import mapAnnotationColorPropertyToPalette from 'constants/mapAnnotationColorPropertyToPalette';
-
-export default store => (toolName, colorPalette) => {
-  const availablePalettes = getDataWithKey(mapToolNameToKey(toolName)).availablePalettes;
-  const property = mapPaletteToAnnotationColorProperty[colorPalette];
-  if (availablePalettes.includes(property)) {
-    store.dispatch(actions.setColorPalette(mapToolNameToKey(toolName), property));
-  } else {
-    console.warn(`${toolName} does not have ${colorPalette} color, available colors are: ${availablePalettes.map(palette => mapAnnotationColorPropertyToPalette[palette]).join(', ')} `);
-  }
-};
\ No newline at end of file
+import { mapToolNameToKey } from 'constants/map';
+import actions from 'actions';
+import selectors from 'selectors';
+
+/**
+ * @typedef WebViewerInstance.PaletteOption
+ * @property {string[]} toolNames Tools that will have the same colors in the palette.
+ * @property {string[]} colors An array of hex colors. Use 'transparency' for a transparent color.
+ */
+
+/**
+ * Sets the colors in the palette globally or for specific tools and their associated annotations
+ * @method WebViewerInstance#setColorPalette
+ * @param {string[]|WebViewerInstance.PaletteOption} An array of hex colors that will override the default colors for every tool.

```

Comment:
are you sure you want to have 2 different kinds of params?
Won't it be a little bit confusing?

What are your thoughts on:
fxn( toolsArray, colorsArray)

?

Reply:
I think I'm ok with the current signature. We have some examples in WebViewer that does this. For example: `loadCanvasAsync`.

Change the signature to be `toolsArray, colorsArray` makes it harder to document. Since the first argument will be `string[]` in both cases, imo it's not very clear to see if it's actually a tool names array or a colors array.

We could make it be `colorsArray, toolsArray`, but I think making it as an object is more explicit.

I think in the future we may use objects as an argument anyways when there's a request for having different colors for different palettes. In this case we could change it to:
```javascript
setColorPalette({
  toolNames: ['...'],
  text: ['...'],
  fill: ['...'],
  border: ['...'],
})
```